pkg/metrics/server.go

```diff
@@ -63,23 +66,27 @@ func stopServer(srv *http.Server) {
 	}
 }
 
+func (Server) Start(ctx context.Context) error {
+	return RunServer(MetricsPort, ctx)
+}
+
 // RunServer starts the server, and watches the tlsCert and tlsKey for certificate changes.
-func RunServer(port int, stopCh <-chan struct{}) {
+func RunServer(port int, ctx context.Context) error {
```

Comment:
The question is if the metrics server provided by the controller runtime will have all the functionality encompassed in the server we have in hand here. Tracking the history in NTO I saw added functionality to the server in hand.
We need to understand if the server here can be replaced completely with the controller runtime one or the diff is too big.
@jmencak  @dagrayvid   this might need to be revisited in this PR or we can refactor the server move in another PR.
At this point of time the metrics reporting is not fully tested (e2e current coverage might not be enough)

Reply:
I use the same metrics, but drop it under the different endpoint, see - https://github.com/openshift/cluster-node-tuning-operator/pull/289/files#diff-b1da0f702ee090192a7506701ce502f780265a647ee3a7b29ac0ce37d2fea1aaR116

Reply:
It seems most operators which use controller-runtime use the kube-rbac-proxy sidecar container to protect the metrics endpoint rather than handling the TLS within the operator. It looks like kube-rbac-proxy does [handle certificate changes](https://github.com/brancz/kube-rbac-proxy/pull/47/files).

I think the cluster-ingress-operator is a good example. The manifests ([servicemonitor](https://github.com/openshift/cluster-ingress-operator/blob/master/manifests/0000_90_ingress-operator_02_servicemonitor.yaml), [deployment](https://github.com/openshift/cluster-ingress-operator/blob/0d5f49acfb118d7a6c553709b730ab1e7b2a6f6c/manifests/02-deployment.yaml#L79-L98), [service](https://github.com/openshift/cluster-ingress-operator/blob/master/manifests/01-service.yaml) look how I would expect. 

I don't fully understand the reasoning for the way the controller-runtime prometheus registry is [combined with the ingress-operator's](https://github.com/openshift/cluster-ingress-operator/blob/0d5f49acfb118d7a6c553709b730ab1e7b2a6f6c/pkg/operator/metrics.go#L13). I was hoping it would be as simple as what is described here https://book.kubebuilder.io/reference/metrics.html#publishing-additional-metrics, but the comment in the ingress operator makes it sound like it might need to be a bit more complicated.